backend/opencv_backend.py

```diff
@@ -0,0 +1,49 @@
+import cv2
+
+from backend import Backend, PublishFunc
+from overlay import Canvas
+
+class OpenCVBackend(Backend):
+    """ Gets and displays video using the OpenCV (`cv2`) library.
+
+        This is intended for use with laptops with webcams, not on the
+        Raspberry Pi. """
+
+    def __init__(self, width: int, height: int, publish_recording_status_func: PublishFunc):
+        super().__init__(width, height, publish_recording_status_func)
+
+        self.webcam = None
+
+        # Time between video frames when running on OpenCV, in milliseconds
+        self.frametime = 17
+
+        self.base_canvas = Canvas(self.width, self.height)
```

Comment:
Create an `OpenCVLayers` enum to reference these layers as this will mimic `PiCameraBackend` as well.

Reply:
I'm kinda hesitant about this, but open to what you think. `picamera` requires whenever you pass the library an image to overlay that you give it some integer layer. On the other hand, `OpenCV` has no such concept of layers - all I'm doing is manually writing one overlay image over the video frame, then the next overlay image, and the next. I don't really know how I'd integrate layer numbers into this process meaningfully. See what is currently called `_on_loop` in this file for the current implementation.

Reply:
![Screen Shot 2020-06-06 at 7 33 56 pm](https://user-images.githubusercontent.com/18709969/83941175-ab0ced80-a82c-11ea-964c-6f06a044a5d9.png)
Does the order of these lines matter? If it does, would it mean that `base_canvas` is 0, `data_canvas` is 1, `message_canvas` = 2? I just think that it allows people to see the order of these layers more clearly? But maybe this is only useful later if we decide to have more canvas 'layers' for the openCV implementation (which is not necessary for the bike atm).

I'm open to the idea so happy with what you feel best!

Reply:
Order doesn't really matter in the current overlays, but it would make sense that base always goes below data and message. I also don't think we will need more canvas layers, in fact I feel the message canvas is unnecessary as it is, it's nothing that can't go in the data overlay. If you're open, I think it's best left as is for now (perhaps we could discus in person sometime if you disagree?)

Reply:
Nah I totally agree. Don't think we need more work for opencv and we only have 3 layers anyway. So happy for you to not do this at all!